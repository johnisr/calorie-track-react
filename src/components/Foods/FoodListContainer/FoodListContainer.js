import React from 'react';
import { connect } from 'react-redux';
import FoodList from '../../FoodList/FoodList';
import { addCurrentFood, removeCurrentFood } from '../../../actions/currentFood';
import selectFoodsWithPages from '../../../selectors/foodsWithPages';

export class FoodListContainer extends React.Component {
  handleClick = (food) => {
    if(this.props.currentFood.id !== '') {
      this.props.removeCurrentFood(food);
    }
    this.props.addCurrentFood(food);
  }
  render() {
    return (
      <FoodList
        foods={this.props.foods}
        onClick={this.handleClick}
        buttonText={'Edit'}
      />
    );
  }
}

const mapStateToProps = state => ({
  foods: selectFoodsWithPages(state.foods, state.foodsFilters),
  currentFood: state.currentFood,
});

const mapDispatchToProps = (dispatch) => ({
  addCurrentFood: (food) => dispatch(addCurrentFood(food)),
  removeCurrentFood: (food) => dispatch(removeCurrentFood(food)),
});

export default connect(mapStateToProps, mapDispatchToProps)(FoodListContainer);
