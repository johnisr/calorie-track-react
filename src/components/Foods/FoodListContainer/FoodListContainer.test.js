import React from 'react';
import { shallow } from 'enzyme';
import { FoodListContainer } from './FoodListContainer';
import foodsData from '../../../tests/fixtures/foods';

let foods, currentFood, addCurrentFood, removeCurrentFood, wrapper;
beforeEach(() => {
  foods = foodsData;
  currentFood = foods[1];
  addCurrentFood = jest.fn();
  removeCurrentFood = jest.fn();
  wrapper = shallow(<FoodListContainer
    foods={foods}
    currentFood={currentFood}
    addCurrentFood={addCurrentFood}
    removeCurrentFood={removeCurrentFood}
  />);

});

it ('should render currently', () => {
  expect(wrapper).toMatchSnapshot();
});

it ('should handleClick with nonempty currentFood', () => {
  wrapper.find('FoodList').prop('onClick')(foods[1]);
  expect(removeCurrentFood).toHaveBeenLastCalledWith(foods[1]);
  expect(addCurrentFood).toHaveBeenLastCalledWith(foods[1]);
});

it ('should handleClick empty currentFood', () => {
  const currentFood = { id: '' };
  wrapper = shallow(<FoodListContainer
    foods={foods}
    currentFood={currentFood}
    addCurrentFood={addCurrentFood}
    removeCurrentFood={removeCurrentFood}
  />);
  wrapper.find('FoodList').prop('onClick')(foods[2]);
  expect(removeCurrentFood).not.toHaveBeenCalled();
  expect(addCurrentFood).toHaveBeenLastCalledWith(foods[2]);
});