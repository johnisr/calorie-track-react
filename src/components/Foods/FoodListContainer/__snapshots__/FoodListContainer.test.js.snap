// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render currently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FoodListContainer
    addCurrentFood={[Function]}
    currentFood={
        Object {
            "amount": 100,
            "base": Object {
              "amount": 100,
              "calories": 165,
              "carbohydrates": 0,
              "fat": 3.6,
              "protein": 31,
            },
            "calories": 165,
            "carbohydrates": 0,
            "createdAt": 1000,
            "fat": 3.6,
            "id": "2",
            "name": "Chicken Breast",
            "protein": 31,
            "timesUsed": 1,
            "unit": "grams",
          }
    }
    foods={
        Array [
            Object {
              "amount": 200,
              "calories": 205,
              "carbohydrates": 44,
              "createdAt": 0,
              "fat": 0.4,
              "id": "1",
              "name": "Rice (Jasmine)",
              "protein": 4.2,
              "timesUsed": 0,
              "unit": "grams",
            },
            Object {
              "amount": 100,
              "base": Object {
                "amount": 100,
                "calories": 165,
                "carbohydrates": 0,
                "fat": 3.6,
                "protein": 31,
              },
              "calories": 165,
              "carbohydrates": 0,
              "createdAt": 1000,
              "fat": 3.6,
              "id": "2",
              "name": "Chicken Breast",
              "protein": 31,
              "timesUsed": 1,
              "unit": "grams",
            },
            Object {
              "amount": 24,
              "calories": 163,
              "carbohydrates": 2.5,
              "createdAt": -1000,
              "fat": 14,
              "id": "3",
              "name": "Almonds",
              "protein": 6,
              "timesUsed": 2,
              "unit": "pieces",
            },
          ]
    }
    removeCurrentFood={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "buttonText": "Edit",
      "foods": Array [
        Object {
          "amount": 200,
          "calories": 205,
          "carbohydrates": 44,
          "createdAt": 0,
          "fat": 0.4,
          "id": "1",
          "name": "Rice (Jasmine)",
          "protein": 4.2,
          "timesUsed": 0,
          "unit": "grams",
        },
        Object {
          "amount": 100,
          "base": Object {
            "amount": 100,
            "calories": 165,
            "carbohydrates": 0,
            "fat": 3.6,
            "protein": 31,
          },
          "calories": 165,
          "carbohydrates": 0,
          "createdAt": 1000,
          "fat": 3.6,
          "id": "2",
          "name": "Chicken Breast",
          "protein": 31,
          "timesUsed": 1,
          "unit": "grams",
        },
        Object {
          "amount": 24,
          "calories": 163,
          "carbohydrates": 2.5,
          "createdAt": -1000,
          "fat": 14,
          "id": "3",
          "name": "Almonds",
          "protein": 6,
          "timesUsed": 2,
          "unit": "pieces",
        },
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "buttonText": "Edit",
        "foods": Array [
          Object {
            "amount": 200,
            "calories": 205,
            "carbohydrates": 44,
            "createdAt": 0,
            "fat": 0.4,
            "id": "1",
            "name": "Rice (Jasmine)",
            "protein": 4.2,
            "timesUsed": 0,
            "unit": "grams",
          },
          Object {
            "amount": 100,
            "base": Object {
              "amount": 100,
              "calories": 165,
              "carbohydrates": 0,
              "fat": 3.6,
              "protein": 31,
            },
            "calories": 165,
            "carbohydrates": 0,
            "createdAt": 1000,
            "fat": 3.6,
            "id": "2",
            "name": "Chicken Breast",
            "protein": 31,
            "timesUsed": 1,
            "unit": "grams",
          },
          Object {
            "amount": 24,
            "calories": 163,
            "carbohydrates": 2.5,
            "createdAt": -1000,
            "fat": 14,
            "id": "3",
            "name": "Almonds",
            "protein": 6,
            "timesUsed": 2,
            "unit": "pieces",
          },
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
